solr/core/src/java/org/apache/solr/handler/component/HttpShardHandler.java

```diff
@@ -80,6 +94,39 @@ public HttpShardHandler(HttpShardHandlerFactory httpShardHandlerFactory) {
     shardToURLs = new HashMap<>();
   }
 
+  /**
+   * Parse the {@value ShardParams#SHARDS_TOLERANT} param from <code>params</code> as a boolean;
+   * accepts {@value ShardParams#REQUIRE_ZK_CONNECTED} as a valid value indicating <code>false
+   * </code>.
+   *
+   * <p>By default, returns <code>false</code> when {@value ShardParams#SHARDS_TOLERANT} is not set
+   * in <code>
+   * params</code>.
+   */
+  public static boolean getShardsTolerantAsBool(SolrQueryRequest req) {
+    String shardsTolerantValue = req.getParams().get(ShardParams.SHARDS_TOLERANT);
+    if (null == shardsTolerantValue
+        || shardsTolerantValue.trim().equals(ShardParams.REQUIRE_ZK_CONNECTED)) {
```

Comment:
we don't normally trim our params.  I think it's very haphazard to do it on-read (like once we do it here and there, then everywhere we wonder, should we do here too?  a mess IMO)

Reply:
https://issues.apache.org/jira/browse/SOLR-6572

Reply:
SOLR-6572 is for XML, and had a different motivating use-case that doesn't apply here.

Reply:
Sorry, let me be clearer. That .trim() was retained only because it was part of the (recent) changes for that ticket that I received in a merge from main: (https://github.com/apache/solr/pull/2311/files) If you are confident that the change made in SOLR-6572 was in error, we can remove it. I simply trusted the change made by @janhoy

Reply:
So, this param may come from http url param or it may come from default section on a request handler in solrcobfig. If the latter, the `true` string may contain white space if the xml is «indented».

We could not trim() all xml values, since sometimes white space is significant.

Perhaps a more readable code would be to not do plain string comparison here but use a utility method boolCompareTrim() or similar? 

I have not looked into the code base this time around, but I seem to remember there was no obvious clean location to normalize these bool-strings during config parse..

Reply:
I feel the pain too. A question would be, why do we not also trim `REQUIRE_ZK_CONNECTED`?

Reply:
shards.tolerant is boolean. It just is incompatible with partialResults = false. The behavior with respect to REQUIRE_ZK_CONNECTED seems suspicious (silently ignore the value of shards.tolerant), but that's a change for a different ticket.